### PR TITLE
SRV-219 Handle classification from site filtering

### PR DESCRIPTION
**Pull request by PawelPodkalicki**
Any suggestions about refactoring `addClassificationToSiteFiltering` and `processClassificationInFiltering` functions and  are nice to hear.

### CODE REVIEW

File: app/Http/Controllers/Manager/SitesController.php [hunk 1]
```diff
@@ -54,9 +57,145 @@ public function create(Request $request): JsonResponse
...
+        $siteExcludes = $site->site_excludes;
+
+        unset($siteRequires['classification']);
+        unset($siteExcludes['classification']);
+
+
+        $publisherId = $site->user_id;
+        $siteId = $site->id;
+
+        if ($this->isRequiredClassified($input)) {
```
Comment (c3zi) [inline app/Http/Controllers/Manager/SitesController.php]: `isClassifiedPositiveOption()`

File: app/Http/Controllers/Manager/SitesController.php [hunk 2]
```diff
@@ -54,9 +57,145 @@ public function create(Request $request): JsonResponse
...
+            list($requireKeywords, $excludeKeywords) = $this->getClassificationRequireKeywords($publisherId, $siteId);
+
+            $siteRequires['classification'] = $requireKeywords;
+            $siteExcludes['classification'] = $excludeKeywords;
+        }
+        if ($this->isExcludedUnclassified($input)) {
```
Comment (c3zi) [inline app/Http/Controllers/Manager/SitesController.php]: `isNotClassifiedNegative()`

File: app/Http/Controllers/Manager/SitesController.php [hunk 3]
```diff
@@ -54,9 +57,145 @@ public function create(Request $request): JsonResponse
...
+
+        unset($siteRequires['classification']);
+        unset($siteExcludes['classification']);
+
+
+        $publisherId = $site->user_id;
+        $siteId = $site->id;
+
+        if ($this->isRequiredClassified($input)) {
+            list($requireKeywords, $excludeKeywords) = $this->getClassificationRequireKeywords($publisherId, $siteId);
```
Comment (c3zi) [inline app/Http/Controllers/Manager/SitesController.php]: `keywordsForPositiveClassification()`

File: app/Http/Controllers/Manager/SitesController.php [hunk 4]
```diff
@@ -54,9 +57,145 @@ public function create(Request $request): JsonResponse
...
+
+            $siteRequires['classification'] = $requireKeywords;
+            $siteExcludes['classification'] = $excludeKeywords;
+        }
+        if ($this->isExcludedUnclassified($input)) {
+            $excludeKeywords = $this->getClassificationExcludeKeywords($publisherId, $siteId);
```
Comment (c3zi) [inline app/Http/Controllers/Manager/SitesController.php]: `keywordsForNegativeClassification($publisherId, $siteId)`

**Review by c3zi — DISMISSED** [review]
1. I'd suggest to move creation of a classification structure to an external class (TransfersRequest, Middleware).
2. Add a new migration to store checkbox status and use it later (it will remove most of the code inside `processClassificationInFiltering`).

**Comment by PawelPodkalicki** [comment]
Add migration and change function names. Unfortunately, I did not extract middleware part from SitesController, becuase during site creation it's id is not known. It is needed for creating keywords.

**Review by c3zi — APPROVED** [review]

**Pull request merged**